clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -143,15 +143,29 @@ public FetchResponse(Struct struct) {
     public Send toSend(String dest, RequestHeader requestHeader) {
         ResponseHeader responseHeader = new ResponseHeader(requestHeader.correlationId());
 
+        boolean hasThrottleTime = struct.hasField(THROTTLE_TIME_KEY_NAME);
```

Comment:
I agree that there are pros/cons. To be clear though, this PR doesn't change the approach taken right?